drivers/gpio/gpio_esp32.c

```diff
@@ -148,17 +148,27 @@ static int gpio_esp32_config(struct device *dev, int access_op,
 
 	pinmux_pin_set(data->pinmux, pin, PIN_FUNC_GPIO);
 	if (flags & GPIO_PUD_PULL_UP) {
-		pinmux_pin_pullup(data->pinmux, pin, PINMUX_PULLUP_ENABLE);
+		r = pinmux_pin_pullup(data->pinmux, pin, PINMUX_PULLUP_ENABLE);
 	} else if (flags & GPIO_PUD_PULL_DOWN) {
-		pinmux_pin_pullup(data->pinmux, pin, PINMUX_PULLUP_DISABLE);
+		r = pinmux_pin_pullup(data->pinmux, pin, PINMUX_PULLUP_DISABLE);
+	}
+
+	if (r < 0) {
+		return r;
 	}
 
 	if (flags & GPIO_DIR_OUT) {
-		pinmux_pin_input_enable(data->pinmux, pin,
+		r = pinmux_pin_input_enable(data->pinmux, pin,
 					PINMUX_OUTPUT_ENABLED);
+		if (r < 0) {
```

Comment:
This can indeed fail: some pins are input only and they can be configured to be output only.  I'd change the `if` to an assertion, as it's very unlikely that this check will be required on a release build of Zephyr.